audit/audit-gcp.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+# set -x -e
+CNCF_GCP_ORG=758905017065
+
+# gcloud organizations describe $CNCF_GCP_ORG 2>&1
+# ERROR: (gcloud.organizations.describe)
+# User [hh@ii.coop] does not have permission to access organization []
+
+format=json
+echo "# Auditing CNCF CGP Org: ${CNCF_GCP_ORG} #"
+gcloud iam roles list --organization=$CNCF_GCP_ORG --format=$format \
+       > cncf-org.roles.$format
+gcloud organizations get-iam-policy $CNCF_GCP_ORG --format=$format \
+       > cncf-org.policy.$format
+echo "## Iterating over Projects ##"
+gcloud projects list \
+       --filter "parent.id=$CNCF_GCP_ORG" \
+       --format "value(name, projectNumber)" \
+    | while read PROJECT NUM; do \
+    export CLOUDSDK_CORE_PROJECT=$PROJECT
+    echo "### Auditing Project: ${PROJECT} ###"
+    mkdir -p $PROJECT
+    gcloud projects get-iam-policy $PROJECT --format=$format > $PROJECT/policy.$format
+    gcloud iam roles list --project $PROJECT --format=$format > $PROJECT/roles.$format
+    mkdir -p $PROJECT/roles
+    for ROLE_PATH in `gcloud iam roles list --project $PROJECT --format="value(name)"`
+    do
+        ROLE=`basename $ROLE_PATH`
+        gcloud iam roles --project=$PROJECT describe $ROLE \
+               --format=json > $PROJECT/roles/$ROLE.json
+    done
+    echo "#### Iterating over ${PROJECT} Services: ####"
+    mkdir -p $PROJECT/services
+    gcloud services list --filter state:ENABLED --format=$format > $PROJECT/services/enabled.$format
+    for service in `gcloud services list --filter state:ENABLED --format=json \
+                    | jq -r .[].config.name | sed s:.googleapis.com::`
+    do
+        case $service in
```

Comment:
We do a bit of iterating over the services. How far do we want to dig into the configuration of each.

Reply:
I am less concerned about service configs than IAM and access.

Reply:
Understood. Dumping the entire config is a bit out of scope, but I added it the documentation/TODO completeness later.